src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts

```diff
@@ -0,0 +1,185 @@
+    });
+  };
+  const addField = (key: OnChangeObjectKeys) => {
+    setTextAreaWithDatesState(prevState => {
+      const newState = { ...prevState };
+      newState[key].push({
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      });
+      return newState;
+    });
+  };
+
+  const [skillsOptions, setSkillsOptions] = useState([
+    { label: 'Business analysis', value: false },
+    { label: 'Consulting', value: false },
+    { label: 'Estimate', value: false },
+    { label: 'Recruiting', value: false },
+    { label: 'SMM', value: false },
+    { label: 'Copyrighting', value: false },
+    { label: 'UI/UX', value: false },
+    { label: 'Administration', value: false },
+    { label: 'Taxation', value: false },
+    { label: 'Coaching', value: false },
+    { label: 'Full stack', value: false },
+    { label: 'Quality control', value: false },
+    { label: 'Communication', value: false },
+    { label: 'JavaScript', value: false },
+    { label: 'QA Automation', value: false },
+    { label: 'ReactJS', value: false },
+    { label: 'Python', value: false },
+    { label: 'Game dev', value: false },
+    { label: 'Flutter', value: false },
+    { label: 'Node.js', value: false },
+    { label: 'DevOps', value: false },
+    { label: 'Scrum Master', value: false },
+    { label: 'Agile Coach', value: false },
+    { label: 'Project Manager', value: false },
+  ]);
+
+  const onSkillsChange = (index: number) => {
+    setSkillsOptions(prevState => {
+      return prevState.map((e, i) => {
+        if (index === i) {
+          return { ...e, value: !e.value };
+        }
+        return e;
+      });
+    });
+  };
+
+  const [englishOption, setEnglishOption] = useState('Pre-intermediate');
+  const englishOptions = [
+    { label: 'Pre-intermediate', value: 'Pre-intermediate' },
+    { label: 'Intermediate', value: 'Intermediate' },
+    { label: 'Upper-intermediate', value: 'Upper-intermediate' },
+  ];
+  const onEnglishOptionChange = ({ target: { value } }: RadioChangeEvent) => {
+    setEnglishOption(value);
+  };
+
+  const [position, setPosition] = useState('');
+  const [positionError, setPositionError] = useState(false);
+  const onPositionChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.currentTarget.value.length > 25) {
+      setPositionError(true);
+    } else {
+      setPosition(e.currentTarget.value);
+      setPositionError(false);
+    }
+  };
+
+  const [category, setCategory] = useState('');
+  const onCategoryChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setCategory(e.currentTarget.value);
+  };
+
+  const [price, setPrice] = useState<number>(0);
+  const onPriceChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (+e.currentTarget.value < 0) {
+      window.alert('Wrong price');
+    } else {
+      setPrice(+e.currentTarget.value);
+    }
+  };
+
+  const [description, setDescription] = useState('');
```

Comment:
types
move all use state together to the top, then useEffects and after onClick and other functions

Reply:
as long as state is not reusable in many places it makes more sense to keep them next to the logic. easier to read